libbeat/cfgfile/list.go

```diff
@@ -87,11 +87,11 @@ func (r *RunnerList) Reload(configs []*reload.ConfigWithMeta) error {
 		wg.Add(1)
 		r.logger.Debugf("Stopping runner: %s", runner)
 		delete(r.runners, hash)
-		go func() {
+		go func(runner Runner) {
 			defer wg.Done()
 			runner.Stop()
 			r.logger.Debugf("Runner: '%s' has stopped", runner)
-		}()
+		}(runner)
```

Comment:
> I wish the linter would have caught this...
> 
> Am I correct that this means some running modules will never actually be stopped depending on the timing? This definitely feels like something we should include in the 8.3.3 release.

Yes, it does. Without this PR, the situation is worse then the original bug :cry: 

But the original buggy PR is very recent, so I strongly believe no releases were made in the mean time.

Reply:
> I wish the linter would have caught this...

Me too, there should be a linter for that. It's such a basic case of variable shadowing...

Reply:
Go vet can do it but it is disabled by default: https://golangci-lint.run/usage/linters/#govet

Probably it will be noisy but I can try turning it on and see what result it gives me.

Reply:
I tried to run `go vet` but for  some reason it didn't detect it. I even tried `go vet -loopclosure`, still not error shown :/

But I didn't spend much time trying to get it working

Reply:
> I tried to run `go vet` but for some reason it didn't detect it. I even tried `go vet -loopclosure`, still not error shown :/

https://cs.opensource.google/go/x/tools/+/refs/tags/v0.1.7:go/analysis/passes/loopclosure/loopclosure.go;l=24;bpv=1 (thanks to gopher slack tools people).